src/kit_version.rs

```diff
@@ -0,0 +1,207 @@
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::de::{self, Deserialize, Deserializer, Visitor};
+use serde::ser::{Serialize, Serializer};
+use std::convert::TryFrom;
+use std::fmt;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct BuildInfo {
+    pub commit: String,
+    pub branch: Option<String>,
+}
+
+impl fmt::Display for BuildInfo {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.branch.as_ref() {
+            None => write!(f, "{}", self.commit),
+            Some(branch) => write!(f, "{}@{}", self.commit, branch),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct KitVersion {
+    pub epoch: u16,
+    pub major: u8,
+    pub minor: u8,
+    pub patch: u8,
+    pub dev: bool,
+    pub build_info: Option<BuildInfo>,
+}
+
+impl fmt::Display for KitVersion {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.dev {
+            match &self.build_info {
+                Some(build_info) => write!(
+                    f,
+                    "{}.{}.{}.{}dev:{}",
+                    self.epoch, self.major, self.minor, self.patch, build_info
+                ),
+                None => write!(
+                    f,
+                    "{}.{}.{}.{}dev",
+                    self.epoch, self.major, self.minor, self.patch
+                ),
+            }
+        } else {
+            match &self.build_info {
```

Comment:
Suggstion: Instead of wrapping the `match` in an `if`, when the `match` is identical, it might be easier to merge them into 1

[Guards](https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#extra-conditionals-with-match-guards) may be useful here.

Reply:
6b5fb90